frontend/app/src/App.tsx

```diff
@@ -739,9 +739,17 @@ export class App extends PureComponent<Props, State> {
   }
 
   handleLogo = (logo: Logo, metadata: ForwardMsgMetadata): void => {
+    // Pass the current page & run ID for cleanup
+    const { scriptRunId } = this.state
+    const { activeScriptHash } = metadata
+
     this.setState(
       {
-        elements: this.pendingElementsBuffer.appRootWithLogo(logo, metadata),
+        elements: this.pendingElementsBuffer.appRootWithLogo(
+          logo,
+          activeScriptHash,
+          scriptRunId
```

Comment:
Not strongly held opinion, but it seems like `activeScriptHash` and `scriptRunId` should be in an object representing the `metadata` for the logo. I think it will just be cleaner. I think it goes well with the function title (appRootWithLogo takes in a logo and associated metadata) and future-proofs as we add more metadata associated with the logo.

Reply:
Think thats fair - made an update let me know if it makes sense to you or not!